### ISSUE TITLE
Accuracy not increasing with a modified version of VGG-16

### ISSUE THREAD

**Issue by sauravtii** [fangwei123456/spikingjelly#303]
Hi, I am using the following architecture and trying to train it using Adamax optimizer. But the accuracy is not at all increasing (the model is learning nothing). Can anyone pelase help me on this?

```
class CNN(nn.Module):
    def __init__(self):
        super().__init__()
        self.network = nn.Sequential(
            nn.Conv2d(3, 4, 3, 1, 1),
            # nn.BatchNorm2d(64),
            nn.ReLU(),

            nn.Conv2d(4, 64, 3, 1, 1),
            # nn.BatchNorm2d(64),
            nn.ReLU(),
            
            nn.Conv2d(64, 64, 3, 1, 1),
            # nn.BatchNorm2d(64),
            nn.ReLU(),
            # nn.AvgPool2d((2,2)),

            nn.Conv2d(64, 64, 3, 2, 1),
            # nn.BatchNorm2d(64),
            nn.ReLU(),

            nn.Conv2d(64, 128, 3, 1, 1),
            # nn.BatchNorm2d(128),
            nn.ReLU(),
            
            nn.Conv2d(128, 128, 3, 1, 1),
            # nn.BatchNorm2d(128),
            nn.ReLU(),
            # nn.AvgPool2d((2,2)),

            nn.Conv2d(128, 128, 3, 2, 1),
            # nn.BatchNorm2d(128),
            nn.ReLU(),

            nn.Conv2d(128, 256, 3, 1, 1),
            # nn.BatchNorm2d(256),
            nn.ReLU(),
            
            nn.Conv2d(256, 256, 3, 1, 1),
            # nn.BatchNorm2d(256),
            nn.ReLU(),
            
            nn.Conv2d(256, 256, 3, 1, 1),
            # nn.BatchNorm2d(256),
            nn.ReLU(),
            # nn.AvgPool2d((2,2)),

            nn.Conv2d(256, 256, 3, 2, 1),
            # nn.BatchNorm2d(256),
            nn.ReLU(),
            
            nn.Conv2d(256, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            
            nn.Conv2d(512, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            
            nn.Conv2d(512, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            # nn.AvgPool2d((2,2)),

            nn.Conv2d(512, 512, 3, 2, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            
            nn.Conv2d(512, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            
            nn.Conv2d(512, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            
            nn.Conv2d(512, 512, 3, 1, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),
            # nn.AvgPool2d((2,2)),

            nn.Conv2d(512, 512, 3, 2, 1),
            # nn.BatchNorm2d(512),
            nn.ReLU(),            
            
            nn.Flatten(),
            nn.Linear(512, 512),
            nn.ReLU(),
            nn.Linear(512, 10),
            nn.ReLU(),

            
        )

    def forward(self,x):
        x = self.network(x)
        return x
```

**Comment by fangwei123456** [comment]
Hi, your network is deep but without `BatchNorm2d`, which is hard to converge.

**Comment by sauravtii** [comment]
Hi, yes I know but I have to use this architecture as I am working on a project that involves it. Do you have any idea how can I achieve a good performance with this architecture and not using `BatchNorm2d`?
